server/core/src/https/oauth2.rs

```diff
@@ -724,6 +737,37 @@ pub async fn oauth2_preflight_options() -> Response {
         .into_response()
 }
 
+#[serde_as]
+#[derive(Deserialize, Debug, Serialize)]
+pub(crate) struct DeviceFlowForm {
+    client_id: String,
+    #[serde_as(as = "Option<StringWithSeparator::<CommaSeparator, String>>")]
+    scope: Option<BTreeSet<String>>,
+    #[serde(flatten)]
+    extra: HashMap<String, String>,
```

Comment:
Should be BTreeMap as HashMap has extra overheads to build, we should only use HashMap for long-lived structures that benefit from the lookup performance.